### PR TITLE
chore(primitives): `hex.String` basic cleanup

### PR DESCRIPTION
**Pull request by abi87**
MInor cleanups, useful for future larger ones:

- Replaced magic numbers and strings with constants already defined
- Reduced code duplication across some functions
- Simplified UTs using `require`
- Consolidated multiple validations into `IsValidHex`

## Summary by CodeRabbit

- **New Features**
	- Enhanced hex string validation and conversion methods for byte manipulation.
	- Improved JSON unmarshalling for execution types.

- **Bug Fixes**
	- Updated tests to cover additional edge cases and ensure robust error handling for hex conversions.

- **Refactor**
	- Streamlined hex validation logic and removed redundant functions for clarity and maintainability.

- **Tests**
	- Expanded test coverage for various byte manipulation methods and hex string handling.

### CODE REVIEW

File: mod/primitives/pkg/bytes/b.go [hunk 1]
```diff
@@ -50,6 +50,6 @@ func (b *Bytes) UnmarshalText(input []byte) error {
 }
 
 // String returns the hex encoding of b.
-func (b Bytes) String() hex.String {
-	return hex.FromBytes(b)
+func (b Bytes) String() string {
```
Comment (abi87) [inline mod/primitives/pkg/bytes/b.go:53]: with this change hex.String is type is called in `string.go` and `hex_test.go` only

File: mod/primitives/pkg/common/consensus.go [hunk 2]
```diff
@@ -84,7 +84,7 @@ func NewRootFromBytes(input []byte) Root {
 }
 
 // Hex converts a root to a hex string.
-func (r Root) Hex() string { return string(hex.EncodeBytes(r[:])) }
```
Comment (abi87) [inline mod/primitives/pkg/common/consensus.go:87]: casting never helps refactoring IMHO.

File: mod/primitives/pkg/encoding/hex/format.go [hunk 3]
```diff
@@ -20,14 +20,11 @@
 
 package hex
 
-// has0xPrefix returns true if s has a 0x prefix.
-func has0xPrefix[T []byte | string](s T) bool {
-	return len(s) >= 2 && s[0] == '0' && (s[1] == 'x' || s[1] == 'X')
-}
```
Comment (abi87) [inline mod/primitives/pkg/encoding/hex/format.go:26]: absorbed into `IsValidHex` to consolidate verifications (which makes them less error prone)

File: mod/primitives/pkg/encoding/hex/format.go [hunk 4]
```diff
@@ -73,13 +71,11 @@ func formatAndValidateText(input []byte) ([]byte, error) {
...
 	// realistically, this shouldn't rarely error if called on
 	// unwrapped hex.String
-	if len(input) == 0 {
-		return *new(T), ErrEmptyString
-	}
-	if !has0xPrefix(input) {
-		return *new(T), ErrMissingPrefix
+	if err := IsValidHex(input); err != nil {
+		return *new(T), err
 	}
```
Comment (abi87) [inline mod/primitives/pkg/encoding/hex/format.go:76]: these two checks happens together most often; this is the reason I can consolidate them into `IsValidaTx`

File: mod/primitives/pkg/encoding/hex/string.go [hunk 5]
```diff
@@ -47,41 +48,38 @@ func NewString[T []byte | string](s T) String {
...
+	if len(s) == 0 {
 		return ErrEmptyString
-	} else if !has0xPrefix(str) {
-		return ErrMissingPrefix
 	}
-	return nil
+	if len(s) >= prefixLen &&
+		strings.ToLower(string(s[:prefixLen])) == prefix {
+		return nil
+	}
```
Comment (abi87) [inline mod/primitives/pkg/encoding/hex/string.go:66]: what was `has0xPrefix` implementation

File: mod/primitives/pkg/encoding/hex/string.go [hunk 6]
```diff
@@ -108,19 +106,9 @@ func FromJSONString[B ~[]byte](b B) String {
...
 
-// Has0xPrefix returns true if s has a 0x prefix.
-func (s String) Has0xPrefix() bool {
-	return has0xPrefix[string](string(s))
-}
-
-// IsEmpty returns true if s is empty.
-func (s String) IsEmpty() bool {
-	return len(s) == 0
-}
```
Comment (abi87) [inline mod/primitives/pkg/encoding/hex/string.go:119]: exported just for tests and always checked together. Consolidated into `IsValidHex`

**Review by nidhi-singh02 — APPROVED** [review]

**Review by itsdevbear — APPROVED** [review]
